src/field_theory/abel_ruffini.lean

```diff
@@ -0,0 +1,373 @@
+| inv (α : E) : is_SBR α → is_SBR α⁻¹
+| rad (α : E) (n : ℕ) (hn : n ≠ 0) : is_SBR (α^n) → is_SBR α
+
+variables (E)
+
+/-- The intermediate field of solvable-by-radicals elements -/
+def SBR : intermediate_field F E :=
+{ carrier := is_SBR F,
+  zero_mem' := by { convert is_SBR.base (0 : F), rw ring_hom.map_zero },
+  add_mem' := is_SBR.add,
+  neg_mem' := is_SBR.neg,
+  one_mem' := by { convert is_SBR.base (1 : F), rw ring_hom.map_one },
+  mul_mem' := is_SBR.mul,
+  inv_mem' := is_SBR.inv,
+  algebra_map_mem' := is_SBR.base }
+
+namespace SBR
+
+variables {F} {E} {α : E}
+
+lemma induction (P : SBR F E → Prop)
+(base : ∀ α : F, P (algebra_map F (SBR F E) α))
+(add : ∀ α β : SBR F E, P α → P β → P (α + β))
+(neg : ∀ α : SBR F E, P α → P (-α))
+(mul : ∀ α β : SBR F E, P α → P β → P (α * β))
+(inv : ∀ α : SBR F E, P α → P α⁻¹)
+(rad : ∀ α : SBR F E, ∀ n : ℕ, n ≠ 0 → P (α^n) → P α)
+(α : SBR F E) : P α :=
+begin
+  revert α,
+  suffices : ∀ (α : E), is_SBR F α → (∃ β : SBR F E, ↑β = α ∧ P β),
+  { intro α,
+    obtain ⟨α₀, hα₀, Pα⟩ := this α (subtype.mem α),
+    convert Pα,
+    exact subtype.ext hα₀.symm },
+  apply is_SBR.rec,
+  { exact λ α, ⟨algebra_map F (SBR F E) α, rfl, base α⟩ },
+  { intros α β hα hβ Pα Pβ,
+    obtain ⟨⟨α₀, hα₀, Pα⟩, β₀, hβ₀, Pβ⟩ := ⟨Pα, Pβ⟩,
+    exact ⟨α₀ + β₀, by {rw [←hα₀, ←hβ₀], refl }, add α₀ β₀ Pα Pβ⟩ },
+  { intros α hα Pα,
+    obtain ⟨α₀, hα₀, Pα⟩ := Pα,
+    exact ⟨-α₀, by {rw ←hα₀, refl }, neg α₀ Pα⟩ },
+  { intros α β hα hβ Pα Pβ,
+    obtain ⟨⟨α₀, hα₀, Pα⟩, β₀, hβ₀, Pβ⟩ := ⟨Pα, Pβ⟩,
+    exact ⟨α₀ * β₀, by {rw [←hα₀, ←hβ₀], refl }, mul α₀ β₀ Pα Pβ⟩ },
+  { intros α hα Pα,
+    obtain ⟨α₀, hα₀, Pα⟩ := Pα,
+    exact ⟨α₀⁻¹, by {rw ←hα₀, refl }, inv α₀ Pα⟩ },
+  { intros α n hn hα Pα,
+    obtain ⟨α₀, hα₀, Pα⟩ := Pα,
+    refine ⟨⟨α, is_SBR.rad α n hn hα⟩, rfl, rad _ n hn _⟩,
+    convert Pα,
+    exact subtype.ext (eq.trans ((SBR F E).coe_pow _ n) hα₀.symm) }
+end
+
+theorem is_integral (α : SBR F E) : is_integral F α :=
+begin
+  revert α,
+  apply SBR.induction,
+  { exact λ _, is_integral_algebra_map },
+  { exact λ _ _, is_integral_add },
+  { exact λ _, is_integral_neg },
+  { exact λ _ _, is_integral_mul },
+  { exact λ α hα, subalgebra.inv_mem_of_algebraic (integral_closure F (SBR F E))
+      (show is_algebraic F ↑(⟨α, hα⟩ : integral_closure F (SBR F E)),
+        by exact (is_algebraic_iff_is_integral F).mpr hα) },
+  { intros α n hn hα,
+    obtain ⟨p, h1, h2⟩ := (is_algebraic_iff_is_integral F).mpr hα,
+    refine (is_algebraic_iff_is_integral F).mp ⟨p.comp (X ^ n),
+      ⟨λ h, h1 (leading_coeff_eq_zero.mp _), by rw [aeval_comp, aeval_X_pow, h2]⟩⟩,
+    rwa [←leading_coeff_eq_zero, leading_coeff_comp, leading_coeff_X_pow, one_pow, mul_one] at h,
+    rwa nat_degree_X_pow }
+end
+
+/-- The statement to be proved inductively -/
+def P (α : SBR F E) : Prop := is_solvable (minpoly F α).gal
+
+lemma induction3 {α : SBR F E} {n : ℕ} (hn : n ≠ 0) (hα : P (α ^ n)) : P α :=
```

Comment:
Are these `aux`, or are they intended for use outside this file as well? If so, a more descriptive name would be good. OR at least a docstring (on all the induction principles) that explains which variants exists, and how they differ.

Reply:
I was viewing these induction lemmas as `aux`. I think that the last theorem effectively subsumes these induction lemmas (each of the induction lemmas states that a certain element, which is known to be solvable by radicals, has a solvable Galois group, but this is now a consequence of the last theorem).

Reply:
ok, can you please add a short remark like that to their docstring? Or call them `induction_aux3`, so that it's clear that you want intend it to be an intermediate result.